src/js/player.js

```diff
@@ -2931,6 +2931,45 @@ class Player extends Component {
     if (isEvented(this)) {
       this.trigger('playerreset');
     }
+    this.resetControlBarUI();
+  }
+
+  /**
+   * Reset Control Bar's UI by calling sub-methods that reset
+   * all of Control Bar's components
+   */
+  resetControlBarUI() {
+    this.resetProgressBar();
+    this.resetPlaybackRate();
+    this.resetVolumeBar();
+  }
+
+  /**
+   * Reset tech's progress so progress bar is reset in the UI
+   */
+  resetProgressBar() {
+    this.currentTime(0);
+    if (isNaN(this.duration())) {
```

Comment:
as part of https://github.com/videojs/video.js/pull/5348, duration can now be NaN, and it's probably more accurate as well.

Reply:
Thank you for the input. The reason why i added this is that when the duration is NaN, the remaining-time-display does not change or update. Hence setting it to 0. Is there any other way to fix this?

Reply:
I think it's more correct to set the duration to `NaN`. We should call duration display's update directly here instead.

Reply:
@gkatsev the thing is if I set it to NaN the remaining time display does not change. Even when calling duration display's update directly.

Reply:
From our analysis, PR #5348 would make setting the duration to NaN update the duration display. Therefore, if this PR follows #5348, setting the duration to NaN will trigger the update in the remaining time display

Reply:
#5348 is in 7.4.0 and 7.4 is about to be promoted to latest. I can confirm that calling `updateContent` on the duration display after setting the duration to `NaN` will change the duration display to be `-:-`. However, the remaining time display does a null check on the duration. This probably needs to be updated to verify that the duration is a number. So, https://github.com/videojs/video.js/blob/f0ba1f5b2affb227cbe69149b2abcfbad8469d16/src/js/control-bar/time-controls/remaining-time-display.js#L64-L66
probably should be something like:
```js
if (typeof this.player_.duration() !== 'number') {
  return;
}
```

Reply:
Allright, we will also add that change to the PR then. It should be submitted in a couple of hours. Thank you for all the help :)

Reply:
@gkatsev I think it's done. Anything else please let me know!